### PR TITLE
[CALCITE-6489] Add more data types to the Arrow test set

### PR DESCRIPTION
**Pull request by caicancai**
https://issues.apache.org/jira/browse/CALCITE-6489

### CODE REVIEW

**Comment by caicancai** [comment]
@asolimando I created a new test data set, which will be used to test arrow type later.

File: arrow/src/test/java/org/apache/calcite/adapter/arrow/ArrowData.java [hunk 1]
```diff
@@ -174,6 +196,53 @@ public void writeArrowData(File file) throws IOException {
     fileOutputStream.close();
   }
 
+  private void populateTypeMetaDataVector(FieldVector vector, int numRows) {
+    switch (vector.getMinorType()) {
+    case TINYINT:
+      byteField(vector, numRows);
```
Comment (XuQianJin-Stars) [inline arrow/src/test/java/org/apache/calcite/adapter/arrow/ArrowData.java:202]: change to `tinyIntField`?

File: arrow/src/test/java/org/apache/calcite/adapter/arrow/ArrowData.java [hunk 2]
```diff
@@ -174,6 +196,53 @@ public void writeArrowData(File file) throws IOException {
...
+    tinyIntVector.allocateNew();
+    for (int i = 0; i < rowCount; i++) {
+      tinyIntVector.set(i, this.byteValue);
+      this.byteValue++;
+    }
+    fieldVector.setValueCount(rowCount);
+  }
+
+  private void smallIntFiled(FieldVector fieldVector, int rowCount) {
+    SmallIntVector shortVector = (SmallIntVector) fieldVector;
```
Comment (XuQianJin-Stars) [inline arrow/src/test/java/org/apache/calcite/adapter/arrow/ArrowData.java:236]: `shortVector` -> `smallIntVector`?

File: arrow/src/test/java/org/apache/calcite/adapter/arrow/ArrowData.java (hunk 1 above)
Comment (caicancai) [reply arrow/src/test/java/org/apache/calcite/adapter/arrow/ArrowData.java]: done

File: arrow/src/test/java/org/apache/calcite/adapter/arrow/ArrowData.java (hunk 2 above)
Comment (caicancai) [reply arrow/src/test/java/org/apache/calcite/adapter/arrow/ArrowData.java]: done

File: arrow/src/test/java/org/apache/calcite/adapter/arrow/ArrowData.java [hunk 3]
```diff
@@ -89,6 +96,26 @@ private Schema makeArrowSchema() {
     return new Schema(childrenBuilder.build(), null);
   }
 
+  private Schema makeArrowTypeMetaDataSchema() {
+    ImmutableList.Builder<Field> childrenBuilder = ImmutableList.builder();
+    FieldType tinyIntType = FieldType.nullable(new ArrowType.Int(8, true));
+    FieldType smallIntType = FieldType.nullable(new ArrowType.Int(16, true));
+    FieldType intType = FieldType.nullable(new ArrowType.Int(32, true));
+    FieldType stringType = FieldType.nullable(new ArrowType.Utf8());
+    FieldType floatType =
```
Comment (mihaibudiu) [inline arrow/src/test/java/org/apache/calcite/adapter/arrow/ArrowData.java:105]: does arrow support both double and single precision for doubles?
Then I would do both at the same time.

**Pull request closed without merging**